packages/core/test/unit/utils/collection-ttl.test.ts

```diff
@@ -19,6 +19,18 @@ describe('CollectionTtl', () => {
       expect(cttl.refreshTtl()).toBe(true);
       expect(cttl.ttlSeconds()).toEqual(ttl);
     });
+
+    it('does not allow negative ttl', () => {
+      expect(() => {
+        CollectionTtl.of(-1);
+      }).toThrow('ttl must be a positive integer');
+    });
+
+    it('does not allow float ttl', () => {
+      expect(() => {
+        CollectionTtl.of(1.5);
+      }).toThrow('ttl must be a positive integer');
+    });
```

Comment:
previously allowed floating point collectionTtl, should that still be the case?

Reply:
Yeah previously the SDK allowed it but it fails at the gRPC serialization layer which surfaces as an `INTERNAL_ERROR`. There's an interesting case though for which I'll tag @cprice404 again as I am not sure why its happening.

A floating point upto three decimal places is being accepted. So

```
await momento.set('cache', 'foo', 'FOO', {ttl: 30.411});
```

succeeds, but

```
await momento.set('cache', 'foo', 'FOO', {ttl: 30.4115});
```

fails with an `Request message serialization failure: Assertion failed` error. 

Same happens with `CollectionTtl` as well. Either way, I don't think a decimal ttl makes seconds and we call it as `ttlSeconds` in a bunch of places so I'd go ahead with what you have in the PR.

Reply:
oh..... I think I understand the reason why this is failing after 3 decimal places. I think the wire format transports the value as a number of milliseconds, not a number of seconds. So there is probably a `* 1000` in the code before we serialize.

We may have a user ask us to support millis. But I think for now let's just do what you are doing in this PR @anitarua , and force an integer. If someone requests support for millis we'll just have to change your validation function to do `* 1000` and then the integer check afterward.

Reply:
maybe we should leave a comment to this effect in the validateTtl function to make it obvious to the next developer if this does come up again.